RecoParticleFlow/PFClusterProducer/plugins/CorrectedECALPFClusterProducer.cc

```diff
@@ -83,56 +72,57 @@ DEFINE_FWK_MODULE(CorrectedECALPFClusterProducer);
-  std::auto_ptr<reco::PFCluster::EEtoPSAssociation> association_out;
-  association_out.reset(new reco::PFCluster::EEtoPSAssociation);
+  std::unique_ptr<reco::PFClusterCollection> clusters_out(new reco::PFClusterCollection);    
+  std::unique_ptr<reco::PFCluster::EEtoPSAssociation> association_out(new reco::PFCluster::EEtoPSAssociation);
   
   edm::Handle<reco::PFClusterCollection> handleECAL;
   e.getByToken(_inputECAL,handleECAL);
   edm::Handle<reco::PFClusterCollection> handlePS;
   e.getByToken(_inputPS,handlePS);
-  
-  clusters_out->reserve(handleECAL->size());
-  association_out->reserve(handleECAL->size());
+
+  auto const & ecals = *handleECAL;
+  auto const & pss = *handlePS;
+
+  clusters_out->reserve(ecals.size());
+  association_out->reserve(ecals.size());
   clusters_out->insert(clusters_out->end(),
-		       handleECAL->begin(),handleECAL->end());
+		       ecals.begin(),ecals.end());
   //build the EE->PS association
-  double dist = -1.0, min_dist = -1.0;
-  for( unsigned i = 0; i < handlePS->size(); ++i ) {      
-    switch( handlePS->at(i).layer() ) { // just in case this isn't the ES...
+  for( unsigned i = 0; i < pss.size(); ++i ) {      
+    switch( pss[i].layer() ) { // just in case this isn't the ES...
     case PFLayer::PS1:
     case PFLayer::PS2:
       break;
     default:
       continue;
     }    
-    edm::Ptr<reco::PFCluster> psclus(handlePS,i);
-    if( psclus->energy() < _minimumPSEnergy ) continue;
-    edm::Ptr<reco::PFCluster> eematch,eeclus;
-    dist = min_dist = -1.0; // reset
-    for( size_t ic = 0; ic < handleECAL->size(); ++ic ) {
-      if( handleECAL->at(ic).layer() != PFLayer::ECAL_ENDCAP ) continue;
-      eeclus = edm::Ptr<reco::PFCluster>(handleECAL,ic);	
-      dist = testPreshowerDistance(eeclus,psclus);      
-      if( dist == -1.0 || (min_dist != -1.0 && dist > min_dist) ) continue;
-      if( dist < min_dist || min_dist == -1.0 ) {
-	eematch = eeclus;
+    if(  pss[i].energy() < _minimumPSEnergy ) continue;
+    int eematch = -1;
+    auto min_dist = std::numeric_limits<double>::max();
+    for( size_t ic = 0; ic < ecals.size(); ++ic ) {
+      if( ecals[ic].layer() != PFLayer::ECAL_ENDCAP ) continue;
+      auto dist = testPreshowerDistance(ecals[ic],pss[i]);      
+      if (dist == -1.0) dist=std::numeric_limits<double>::max(); 
+      if( dist < min_dist ) {
+	eematch = ic;
 	min_dist = dist;
       }
     } // loop on EE clusters      
-    if( eematch.isNonnull() ) {      
-      association_out->push_back(std::make_pair(eematch.key(),psclus));
+    if( eematch>=0 ) {
+      edm::Ptr<reco::PFCluster> psclus(handlePS,i);
+      association_out->push_back(std::make_pair(eematch,psclus));
     }
   }
   std::sort(association_out->begin(),association_out->end(),sortByKey);
   
   _corrector->correctEnergies(e,es,*association_out,*clusters_out);
   
   association_out->shrink_to_fit();
+
+  // std::cout << "ass " << association_out->size() << std::endl;
```

Comment:
was there for fast regression test in HLT (I have a version of 8_1_0_pre1 will all these cout in)